### PR TITLE
Switch from Travis to Github Actions

### PR DESCRIPTION
**Pull request by Bensuo**
Moves CI testing from Travis to Github Actions, given the impending shutdown of Travis' `.org` service.

Changes:

- Adds a Github Actions workflow which replicates the original Travis workflow. Runs on pushes to `master` as well as pull requests and on-demand through the Actions tab.
- Adds a `Dependabot` config to track updates to Actions dependancies.
- Removes the `.travis.yml` file.
- Fixed missing SDK logo in `README.md` and replaced Travis status badge with Actions badge.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
-# ComputeCpp SDK Readme
 
-![ComputeCpp™]
+<div align="center">
```
Comment (DuncanMcBain) [inline README.md:2]: Is there no way to link this new image in the old style? If you check the bottom of the document, that's where we've defined the links to keep the main body a bit more readable. At the very least, the old links should be removed from there, though I'd prefer that we add the links there instead.

File: README.md [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+  <br /><img src="https://developer.codeplay.com/assets/img/home/tools/computecppsdk.png"><br />
+</div>
 
+# ComputeCpp SDK Readme
+[![Build and Test](https://github.com/codeplaysoftware/computecpp-sdk/actions/workflows/build-test.yml/badge.svg)](https://github.com/codeplaysoftware/computecpp-sdk/actions/workflows/build-test.yml)
```
Comment (DuncanMcBain) [inline README.md:7]: Similarly, I'd hope that we can update the travis SVG link at the end of the file to the new shiny actions one you have here, which keeps things a bit more organised (while still leaving a useful link inline).

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: daily
```
Comment (DuncanMcBain) [inline .github/dependabot.yml:6]: Will this build the SDK once per day? If so, I'd not bother. Provided it builds master when it's updated, that should be fine. (What do we gain from daily builds?)

**Review by DuncanMcBain — CHANGES_REQUESTED** [review]
I can't leave a comment on it, but I'd probably keep the `build_computecpp.sh` file in the `.github` folder or similar. It's not used anywhere else, which is why it was in `.travis` before.

The actions stuff looks really good though!

File: .github/dependabot.yml (hunk 3 above)
Comment (DuncanMcBain) [reply .github/dependabot.yml:6]: I see an answer to this elsewhere!

File: README.md (hunk 1 above)
Comment (Bensuo) [reply README.md]: Ah that is my mistake, I totally missed the links being defined at the end of the file! I've reverted this to the old style and updated the link for it at the bottom of the document.

File: README.md (hunk 2 above)
Comment (Bensuo) [reply README.md]: I've also reverted this to the old style and updated the links!

**Comment by Bensuo** [comment]
> I can't leave a comment on it, but I'd probably keep the `build_computecpp.sh` file in the `.github` folder or similar. It's not used anywhere else, which is why it was in `.travis` before.

The script has now been moved to the `.github` folder!

**Review by DuncanMcBain — APPROVED** [review]

**Comment by DuncanMcBain** [comment]
Please use squash + rebase for PRs to this repo in future, thanks!

**Pull request merged**